gateway-framework/src/network/exchange_rate.rs

```diff
@@ -35,11 +34,10 @@ pub fn grt_per_usd(provider: Url) -> anyhow::Result<Eventual<NotNan<f64>>> {
         ChainlinkPriceFeed::new(chainlink_usd_per_eth, provider),
     ));
 
-    let (writer, reader) = Eventual::new();
-    let writer: &'static Mutex<EventualWriter<NotNan<f64>>> =
-        Box::leak(Box::new(Mutex::new(writer)));
-    eventuals::timer(Duration::from_secs(60))
-        .pipe_async(move |_| async {
+    let (tx, mut rx) = watch::channel(NotNan::new(0.0).unwrap());
+    tokio::spawn(async move {
+        loop {
+            sleep(Duration::from_secs(60)).await;
```

Comment:
I just noticed that putting the sleep at the beginning of the loop will cause an initial delay of 60 seconds to fetch the first update. This is not desirable. I'd rather put it at the end of the loop.

Reply:
https://github.com/edgeandnode/gateway/pull/730